mutation_partition.cc

```diff
@@ -2038,13 +2038,63 @@ reconcilable_result reconcilable_result_builder::consume_end_of_stream() {
                                std::move(_memory_accounter).done());
 }
 
+template<FlattenedConsumer Delegate>
+class position_recording_consumer {
+    Delegate _delegate;
+    std::optional<dht::decorated_key> _pkey;
+    std::optional<clustering_key> _ckey;
+public:
+    position_recording_consumer(Delegate delegate)
+        : _delegate(std::move(delegate))
+    { }
+
+    void consume_new_partition(const dht::decorated_key& dk) {
+        _pkey = dk;
+        _ckey = std::nullopt;
+        _delegate.consume_new_partition(dk);
+    }
+
+    void consume(tombstone t) {
+        _delegate.consume(std::move(t));
+    }
```

Comment:
Partition tombstone is always emitted after partition start and there is no way to stop consuming in such a way that you consume the partition start but not the tombstone. We can only stop later in the partition.

Regarding storing rows, not sure I correctly understand the question. So let me paraphrase. The question is why do we need to record the clustering key in position_recording_consumer if we already consume the rows and put them in the query result, why can't we get the key from query result? The answer is that query result which the compacting consumer builds may not contain any rows, e.g. because they are covered by the partition tombstone. We need to record the position up to which the consumer consumed, not the position in the output.